dbt-databricks adapter documentation. (#1092)
fishtown-analytics/docs.getdbt.com: website/docs/reference/warehouse-profiles/databricks-profile.md

---
title: "Databricks Profile"
id: "databricks-profile"
---

> ## Overview of dbt-spark

Reviewer: ```suggestion
## Overview of dbt-databricks
```

> **Maintained by:** some dbt loving bricksters

Reviewer 2: nit on casing, should be `Bricksters`.

**Author:** Databricks
**Source:** [Github](https://github.com/databricks/dbt-databricks)    
**dbt Cloud:** Coming Soon    
**dbt Slack channel** [Link to channel](https://getdbt.slack.com/archives/CNGCW8HKL)     

![dbt-databricks stars](https://img.shields.io/github/stars/databricks/dbt-databricks?style=for-the-badge)

## Installation and Distribution

The easiest way to install dbt-databricks is to use `pip`:

```zsh
pip install dbt-databricks
```

### Set up a Databricks Target

> dbt-databricks can connect to a Databricks all-purpose cluster as well as a SQL endpoints.

Reviewer: This is something that I noticed trips customers up a lot. Could you point to any documentation/write a blurb to help clients understand the pros and cons?

> <File name='~/.dbt/profiles.yml'>

Reviewer: Could we provide instructions on where to find the below credentials?

Author: Added a doc reference.

```yaml
your_profile_name:
  target: dev
  outputs:
    dev:
      type: databricks

>       schema: [database/schema name]

Reviewer 2: May be useful to note here that database === schema.

Reviewer: Yes please! We are working to improve our docs to make it less confusing about database/schemas functionalities but until then, clarification is always good.

Author: We've switched all Databricks docs to refer to schema instead of database, so I'll just leave it at schema here to match.

      host: [yourorg.databrickshost.com]
      http_path: [/sql/your/http/path]

>       token: [dapiXXXXXXXXXXXXXXXXXXXXXXX]

Reviewer 2: Might want to note somewhere that this is the Personal Access Token

---
title: "Databricks Profile"
id: "databricks-profile"
---

## Overview of dbt-databricks

**Maintained by:** some dbt loving Bricksters
**Author:** Databricks
**Source:** [Github](https://github.com/databricks/dbt-databricks)
**dbt Cloud:** Coming Soon
**dbt Slack channel** [Link to channel](https://getdbt.slack.com/archives/CNGCW8HKL)

![dbt-databricks stars](https://img.shields.io/github/stars/databricks/dbt-databricks?style=for-the-badge)

## Installation and Distribution

The easiest way to install dbt-databricks is to use `pip`:

```zsh
pip install dbt-databricks
```

### Set up a Databricks Target

dbt-databricks can connect to Databricks all-purpose clusters as well as SQL endpoints.

> The later provides an opioniated way of running SQL workloads with optimal performance and

Reviewer: ### 
```suggestion
The latter provides an opinionated way of running SQL workloads with optimal performance and
```

price, the former provides all the flexibility of Spark.

<File name='~/.dbt/profiles.yml'>

```yaml
your_profile_name:
  target: dev
  outputs:
    dev:
      type: databricks
      schema: [schema name]
      host: [yourorg.databrickshost.com]
      http_path: [/sql/your/http/path]
      token: [dapiXXXXXXXXXXXXXXXXXXXXXXX] # Personal Access Token (PAT)
```

</File>

See the [Databricks documentation](https://docs.databricks.com/dev-tools/dbt.html#) on how
to obtain the credentials for configuring your profile.

## Caveats

### Supported Functionality

Most dbt Core functionality is supported, but some features are only available
on Delta Lake.

Delta-only features:
1. Incremental model updates by `unique_key` instead of `partition_by` (see [`merge` strategy](spark-configs#the-merge-strategy))

> 2. [Snapshots](snapshots)

Reviewer: ```suggestion
2. [Snapshots](https://docs.getdbt.com/docs/building-a-dbt-project/snapshots)
```

### Choosing between dbt-databricks and dbt-spark

While `dbt-spark` can be connect to Databricks, `dbt-databricks` was created to make it
even easier to use dbt with the Databricks Lakehouse.

`dbt-databricks` includes:

> - No need to install additional drivers or dependencies

Reviewer: ```suggestion
- No need to install additional drivers or dependencies for use on the CLI
```
